internal/featuresupport/feature_support_level.go

```diff
@@ -122,3 +129,34 @@ func IsFeatureSupported(featureId models.FeatureSupportLevelID, openshiftVersion
 func isArchitectureSupported(featureId models.ArchitectureSupportLevelID, openshiftVersion string) bool {
 	return GetSupportLevel(featureId, openshiftVersion) == models.SupportLevelSupported
 }
+
+func IsFeatureCompatible(feature SupportLevelFeature, features ...SupportLevelFeature) (bool, string) {
+	incompatibilities := feature.GetIncompatibleFeatures()
+	if incompatibilities != nil {
+		for _, f := range features {
+			if funk.Contains(*incompatibilities, f.GetId()) {
+				return false, string(f.GetId())
+			}
+		}
+	}
+
+	return true, ""
+}
+
+func ValidateIncompatibleFeatures(cluster common.Cluster, updateParams *models.V2ClusterUpdateParams) error {
+	var activatedFeatures []SupportLevelFeature
+
+	for _, feature := range featuresList {
+		if feature.IsFeatureActivated(cluster, updateParams) {
+			activatedFeatures = append(activatedFeatures, feature)
+		}
+	}
+
+	for _, feature := range activatedFeatures {
+		if isCompatible, incompatibleFeature := IsFeatureCompatible(feature, activatedFeatures...); !isCompatible {
+			return fmt.Errorf("cannot activate %s feature alnogside %s feature", feature.GetId(), incompatibleFeature)
```

Comment:
if the feature is not supported for specific architecture - the message should be different and align with all error messages in the code 
"return errors.New("only x86-64 CPU architecture is supported on Nutanix clusters")"
so I think we should split this method and first check if it's compatible with the architecture - if not - return specific message 
if yes - return other message 
@nirfarkas WDYT?